synapse/handlers/room.py

```diff
@@ -1008,6 +1008,7 @@ async def get_event_context(
         event_id: str,
         limit: int,
         event_filter: Optional[Filter],
+        use_admin_priviledge: bool = False,
```

Comment:
I think for paranoia's sake we should replace the `user: UserID` param with a `requester: Requester` param, that way we can asset that if `use_admin_priviledge` is set then the requester is a server admin.